### PR TITLE
Solve macOS problems in test_async_asyncio.py

### PR DESCRIPTION
**Pull request by janiversen**
The failing tests mentioned in
https://github.com/riptideio/pymodbus/discussions/729
is solved. It was mostly a matter of a missing "async" keyword.

However with this PR it became clear, that there are a loose task (the event loop) that are never closed, that is however for another PR.

### CODE REVIEW

**Review by dhoomakethu — APPROVED** [review]

**Comment by janiversen** [comment]
@dhoomakethu since you approved this PR (and hopefully soon merged), the discussion can be closed, but seems only maintainers can do that.

**Comment by dhoomakethu** [comment]
@janiversen I can add you as a maintainer if you are interested.

**Comment by janiversen** [comment]
It would surely make things easier and I can help a lot more, so I would be happy to accept that.

**Comment by dhoomakethu** [comment]
Btw , can you raise the PR against `dev` branch. I still would like to go from `dev` to `3.0.0` branch.

**Comment by janiversen** [comment]
Thanks a lot, I will use it with care. In general I prefer to have a second opinion on my PRs, unless they are very trivial, so I will continue to ask for a review.

I will do the rebase…does does contain the current dev3.x so I can rebase all 3.0 work to dev ?

**Comment by dhoomakethu** [comment]
dev is almost in sync with 3.0.0. So should be fine to rebase.

**Pull request closed without merging**
